### PR TITLE
Events

### PR DESCRIPTION
**Pull request by tahiry-dev**
-Create a "sign-up system" which allow user to create an account and logging

-Create simple association between tables and Adjust the view accordingly

-Configure and code the App to allow users follow events

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(params.require(:user).permit(:name))
+    if @user.save
+      log_in @user
+      redirect_to @user
+    else
+      render :new
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: When a user logs in, they should be redirected to the new Event path.

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :creator, class_name: 'User'
+  validates :description, presence: true, length: { minimum: 6 }
+  validates :date, presence: true
+  validates :title, presence: true
+
+  has_many :attendances, foreign_key: :attended_event_id, dependent: :destroy
+  has_many :attendees, through: :attendances
+
+  scope :past, -> { where('date<?', Time.now) }
+  scope :future, -> { where('date>?', Time.now) }
```
Comment (peterrobert) [inline app/models/event.rb]: This should be greater or equal to.  Kindly check below. ⬇️ 
`scope :upcoming, -> { where('date >= ?', Time.zone.now) }`

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED ♻️ 🛑 

Hello @tahiry-dev, very nice work. You have done such an impressive job 👏🏿 I love that you have implemented almost all the requirements for this project. Just a few changes and we shall be good to go kindly check below ⬇️ 

- [ ] All navigation links, should be present at the top of the page. 
navigation_links_requirement

if you have any question kindly reach me on slack @peterrobert

### HAPPY CODING 💻

**Comment by tahiry-dev** [comment]
Thanks for those useful reviews

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] You shouldn't be able to see the link to `login` when you are already logged in
- [ ] You don't have a user show page link. To get to the user show page, I had to type it out in the address bar myself. This can be implemented by displaying the username of the current-signed-in-user in the navbar and making it a link to the user show page.

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**
I would be happy to help! 👩‍💻 

Happy coding 🤓

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

**Pull request merged**